web_src/js/features/repo-wiki.js

```diff
@@ -53,13 +53,7 @@ async function initRepoWikiFormEditor() {
     previewWiki: true,
     easyMDEOptions: {
       previewRender: (_content, previewTarget) => previewTarget.innerHTML, // disable builtin preview render
-      toolbar: ['bold', 'italic', 'strikethrough', '|',
-        'heading-1', 'heading-2', 'heading-3', 'heading-bigger', 'heading-smaller', '|',
-        'gitea-code-inline', 'code', 'quote', '|', 'gitea-checkbox-empty', 'gitea-checkbox-checked', '|',
-        'unordered-list', 'ordered-list', '|',
-        'link', 'image', 'table', 'horizontal-rule', '|',
-        'clean-block', 'preview', 'fullscreen', 'side-by-side', '|', 'gitea-switch-to-textarea'
-      ],
+      isWiki: true,
```

Comment:
Could we do not use this `isWiki`?

We can just make all buttons as a map, let callers decide which buttons they need.

Otherwise , the `isWiki` creates a strange dependency between the component and it's callers.

Reply:
Caller needs access to `EasyMDE` object to define button actions, so no, not easily.

Reply:
From a framework view, the wiki uses editor, tells the editor which buttons it needs, while the editor shouldn't share wiki's knowledge.

Reply:
Feel free to edit, but from my point of view, current version is sufficient.

Reply:
> Caller needs access to EasyMDE object to define button actions, so no, not easily.

I have done so, that's how the `gitea-...` buttons work.

>  Feel free to edit, but from my point of view, current version is sufficient.

I have no writer permission.

Reply:
> > Caller needs access to EasyMDE object to define button actions, so no, not easily.
> 
> I have done so, that's how the `gitea-...` buttons work.
> 
> > Feel free to edit, but from my point of view, current version is sufficient.
> 
> I have no writer permission.

Now you have, I guess you could somehow map it from string during initialization, yes. Just remember `clean-block`, `heading-bigger`, `heading-smaller` are gone so far.

Reply:
I will try. And I guess it's better to move the EasyMDE action code into a separate file, to keep our editor JS simple, and make the future removal of EasyMDE easier. 

I will try, just a while.

Reply:
The `isWiki` addition also added a bug in the non-wiki init, but I guess you can fix that as well 😆

Reply:
Done in 9e9a4e7a59b9d070b0a48ead38ead9f940c6fca0

Reply:
Works, thanks.